src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -70,25 +69,29 @@ def check_source_access
         options = { use_source: false, follow_project_links: true, follow_multibuild: true }
         src_package = Package.get_by_project_and_name(source_project_name, source_package_name, options)
 
-        raise Pundit::NotAuthorizedError unless PackagePolicy.new(@token.user, src_package).create_branch?
+        raise Token::Errors::CanNotBranchPackageNoPermission, 'Your package could not be branched: not enough permissions.' unless
+          PackagePolicy.new(@token.user, src_package).create_branch?
       end
 
       def branch
         check_source_access
-        BranchPackage.new({ project: source_project_name, package: source_package_name,
-                            target_project: target_project_name,
-                            target_package: target_package_name }).branch
+
+        begin
+          BranchPackage.new({ project: source_project_name, package: source_package_name,
+                              target_project: target_project_name,
+                              target_package: target_package_name }).branch
+        rescue BranchPackage::InvalidArgument, InvalidProjectNameError, ArgumentError => e
```

Comment:
I doubt that [ArgumentError](https://ruby-doc.org/core-2.5.0/ArgumentError.html) is something we should rescue from. Passing the wrong number of arguments is definitely an error in the code, not something that we should expect to happen when branching a package.

Reply:
I added it because of this line:

https://github.com/openSUSE/open-build-service/blob/8f4ebd44d411cc002a6417072bbdacf6467b0e66/src/api/app/models/branch_package.rb#L506

`set_update_project_attribute` is called inside the `branch` method of the BranchPackage model.

Reply:
Oh... what is that code.... Alright... I guess it's fine then :+1: